src/main/webapp/js/finscholar/commonlistview/commonlistview.js

```diff
@@ -0,0 +1,134 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/** @fileoverview The mini controller for scholarship list view. */
+
+goog.module('finscholar.commonlistview');
+
+const {ScholarshipListDataHandler} = goog.require('datahandlers.scholarshiplistdatahandler');
+const {commonlistview, scholarshiplistitems, collegelistitems, loading, endoflist} = goog.require('finscholar.commonlistview.templates');
+const googDom = goog.require('goog.dom');
+
+const EMPTY_STRING = '';
+const ITEM_CONTAINER_ID = 'table-body';
+const STATUS_BAR_ID = 'status';
+
+/** The mini controller for scholarship list view. */
+class CommonListView {
+  
+  constructor(dataHandler, optionIndex) {
+    /** @private @const {ScholarshipListDataHandler | CollegeListDataHandler} */
```

Comment:
Don't allow multiple types. The two dataHandlers should inherit from a base class which guarantees the structure of the classes.